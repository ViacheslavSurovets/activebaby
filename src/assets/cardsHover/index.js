import chantal from './chantal.jpeg';
import cristiano from './cristiano.jpeg';

export {
  chantal,
  cristiano
};
