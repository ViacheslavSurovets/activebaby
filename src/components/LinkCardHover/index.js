export { default as LinkCardHover } from './LinkCardHover';
