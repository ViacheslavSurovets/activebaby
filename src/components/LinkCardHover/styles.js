import styled from 'styled-components';
import { Link } from 'react-router-dom';
import { flexCommon } from '@src/styles/commonStylesAndStyledComponents';
import { mobile, tablet } from '@utils/media';

export const LinkCard = styled ( Link )`
  ${ flexCommon };
  width: 45vw;
  height: 40vw;
  background:url(${ props => props.background });
  background-size: cover;
  text-decoration: none;
  color: ${ ( { theme } ) => theme.main.linkCardHover.linkCardColor };
   ${ tablet }{
    width: 90vw;
    height: 80vw;
  }
  ${ mobile }{
    width: 90vw;
    height: 80vw;
  }
`;

export const LinkCardTitle = styled.span`
  font-size: 4rem;
  background:linear-gradient(currentColor, currentColor) 50% 100% no-repeat;
  background-size:0 1px;
  transform: translateY(1rem);
  transition: all 1s;
`;
export const LinkCardButton = styled.div`
  ${ flexCommon };
  background-color: ${ ( { theme } ) => theme.main.linkCardHover.linkCardButtonBackground };
  height: 6rem;
  min-width: 12rem;
  color: ${ ( { theme } ) => theme.main.linkCardHover.linkCardButtonColor };
  opacity: 0;
  visibility: hidden;
  transition: all .3s;
`;


export const LinkCardContentWrapper = styled.div`
  ${ flexCommon };
  flex-direction: column;
  position:absolute;
  width: 45vw;
  height: 40vw;
  ${ tablet }{
    width: 90vw;
    height: 80vw;
  }
  ${ mobile }{
    width: 90vw;
    height: 80vw;
  }

  &:hover {
    background: rgba(0, 0, 0, .5);
    .js-fakeButton-linkCard {
       opacity: 1;
       visibility: visible;
       transition: all .8s ease-in;
    };
    .js-title-linkCard{
      transform: translateY(-.7rem);
      background-size:100% 2px;
      transition: all .7s;
    }
  }
`;

export const Img = styled.img`
  width: 100%;
  height: 100%;
`;
