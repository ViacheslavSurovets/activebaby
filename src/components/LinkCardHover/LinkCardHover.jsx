import React from 'react';
import Proptypes from 'prop-types';
import { Img, LinkCard, LinkCardButton, LinkCardContentWrapper, LinkCardTitle } from './styles';
import ScrollAnimation from 'react-animate-on-scroll';

const LinkCardHover = ( { data } ) => {
  const { background, title, imageUrl, path } = data;
  return(
    <ScrollAnimation className='js-linkCardHover'  animateIn='fadeIn' animateOnce={true} scrollableParentSelector='#js-layout'>
      <LinkCard to={ path } background={ imageUrl }>
        <Img src={ background } alt={ title } />
        <LinkCardContentWrapper>
          <LinkCardTitle className='js-title-linkCard'>{ title.toUpperCase () }</LinkCardTitle>
          <LinkCardButton className='js-fakeButton-linkCard'>button</LinkCardButton>
        </LinkCardContentWrapper>
      </LinkCard>
    </ScrollAnimation>
  );
};

export default LinkCardHover;

LinkCardHover.propTypes = {
  data: Proptypes.object
};
