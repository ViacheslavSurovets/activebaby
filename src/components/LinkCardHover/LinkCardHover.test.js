import React from 'react';
import { shallow } from 'enzyme';
import LinkCardHover from './LinkCardHover';
import { chantal } from '@assets/cardsHover';


describe ( 'LinkCardHover', () => {
  const mockCardData = {
    id: 1,
    title: 'adamex',
    imageUrl: chantal,
    path: '#'
  };

  it ( 'LinkCardHover Snapshot', () => {
    expect ( shallow ( <LinkCardHover data={ mockCardData } /> ) ).toMatchSnapshot ();
  } );
} );
