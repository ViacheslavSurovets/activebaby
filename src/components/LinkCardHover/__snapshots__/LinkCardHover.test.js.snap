// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkCardHover LinkCardHover Snapshot 1`] = `ShallowWrapper {}`;
